Add AES-256 usage notes to "How does TDE encrypt data?" (#5269)
EnterpriseDB/docs: product_docs/docs/tde/15/index.mdx

### How does TDE encrypt data? 

Starting with version 16, EDB TDE introduces the option to choose between AES-128 and AES-256 encryption algorithms during the initialization of the Postgres cluster. The choice between AES-128 and AES-256 hinges on balancing performance and security requirements. AES-128 is commonly advised for environments where performance efficiency and lower power consumption are pivotal, making it suitable for most applications. Conversely, AES-256 is recommended for scenarios demanding the highest level of security, often driven by regulatory mandates.

> TDE employs AES-128-XTS or AES-256-XTS algorithms for the encryption of data files. The XTS tweak uses the database OID, the relfilenode, and the block number. 

Reviewer: ```suggestion
TDE employs AES-128-XTS or AES-256-XTS algorithms for the encryption of data files. XTS uses a second value, known as the "tweak value", to enhance the encryption. The XTS tweak value with TDE uses the database OID, the relfilenode, and the block number. 
```

> TDE employs AES-128-XTS or AES-256-XTS algorithms for the encryption of data files. XTS uses a second value, known as the "tweak value", to enhance the encryption. The XTS tweak value with TDE uses the database OID, the relfilenode, and the block number. 

Reviewer 2: ```suggestion
TDE uses AES-128-XTS or AES-256-XTS algorithms for encrypting data files. XTS uses a second value, known as the *tweak value*, to enhance the encryption. The XTS tweak value with TDE uses the database OID, the relfilenode, and the block number. 
```

> For Write-Ahead Log (WAL) files, TDE utilizes AES-128-CTR or AES-256-CTR, incorporating the WAL's Log Sequence Number (LSN) as the counter component. 

Reviewer 2: ```suggestion
For write-ahead log (WAL) files, TDE uses AES-128-CTR or AES-256-CTR, incorporating the WAL's log sequence number (LSN) as the counter component. 
```
